src/extension.ts

```diff
@@ -43,6 +45,24 @@ export function activate(context: vscode.ExtensionContext) {
     }
   );
 
+const invokeFullStackReport = (uri : vscode.Uri) => {
+  const fileUri = uri || vscode.window.activeTextEditor.document.uri;
+  multimanifestmodule.dependencyAnalyticsReportFlow(context, fileUri);
+};
+
+const recordAndInvoke = (origin: string, uri : vscode.Uri) => {
+  record(origin);
+  invokeFullStackReport(uri);
+};
+
+const stackAnalysisCommands = [
+  vscode.commands.registerCommand(Commands.TRIGGER_FULL_STACK_ANALYSIS_FROM_EDITOR, recordAndInvoke.bind(null, ActionName.vulnerabilityReportEditor)),
+  vscode.commands.registerCommand(Commands.TRIGGER_FULL_STACK_ANALYSIS_FROM_EXPLORER, recordAndInvoke.bind(null, ActionName.vulnerabilityReportExplorer)),
+  vscode.commands.registerCommand(Commands.TRIGGER_FULL_STACK_ANALYSIS_FROM_PIE_BTN, recordAndInvoke.bind(null, ActionName.vulnerabilityReportPieBtn)),
+  vscode.commands.registerCommand(Commands.TRIGGER_FULL_STACK_ANALYSIS_FROM_STATUS_BAR, recordAndInvoke.bind(null, ActionName.vulnerabilityReportStatusBar)),
+];

```

Comment:
Can this be moved into a separate function?

Reply:
yup

Reply:
registerStackAnalysisCommands()